torch/nn/modules/transformer.py

```diff
@@ -321,25 +321,31 @@ def forward(self, src: Tensor, src_mask: Optional[Tensor] = None, src_key_paddin
         Shape:
             see the docs in Transformer class.
         """
+
+        # see Fig. 1 of https://arxiv.org/pdf/2002.04745v1.pdf
+
+        # self-attention block
+        def mha(x):
+            x = self.self_attn(x, x, x,
+                               attn_mask=src_mask,
+                               key_padding_mask=src_key_padding_mask,
+                               need_weights=False)[0]
+            return self.dropout1(x)
+
+        # feed forward block
+        def ff(x):
+            x = self.linear2(self.dropout(self.activation(self.linear1(x))))
+            return self.dropout2(x)
```

Comment:
Looks like these function definitions break JIT :(

Could try adding them as private functions instead (e.g. `_mha_block()`, `_ffn_block()`)

Reply:
argh. ok made them methods, that seems more orderly than private functions. But maybe there's a reason why private functions would be better? Happy to change if so, my preference is just aesthetic.

Reply:
Ah yeah, sorry I should have said "private methods". I think they're good as methods, but they should be marked as private with the leading underscore so no one depends on them if we want to refactor later on